code/modules/hydroponics/fermenting_barrel.dm

```diff
@@ -0,0 +1,99 @@
+/obj/structure/fermenting_barrel
+	name = "wooden barrel"
+	desc = "A large wooden barrel. You can ferment fruits and such inside it, or just use it to hold liquid."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "barrel"
+	density = TRUE
+	anchored = FALSE
+	container_type = DRAINABLE | AMOUNT_VISIBLE
+	var/open = FALSE
+	pressure_resistance = 2 * ONE_ATMOSPHERE
+	max_integrity = 300
+	var/list/ferment_times = list()
+
+/obj/structure/fermenting_barrel/Initialize()
+	create_reagents(500) //Half of a beer keg, since it can be refilled.
+	. = ..()
+
+/obj/structure/fermenting_barrel/Destroy()
+	QDEL_LIST(ferment_times)
+	. = ..()
+
+/obj/structure/fermenting_barrel/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>It is currently [open?"open, letting you pour liquids in.":"closed, letting you draw liquids from the tap."]</span>")
+
+/obj/structure/fermenting_barrel/process()
+	if(contents.len == 0)
+		STOP_PROCESSING(SSobj,src)
+		return
+
+	for(var/obj/item/reagent_containers/food/snacks/grown/fruit in contents)
+		if(ferment_times["[REF(fruit)]"])
+			ferment_times["[REF(fruit)]"]--
+		else
+			ferment_times -= "[REF(fruit)]"
+			if(fruit.reagents)
+				fruit.reagents.trans_to(src, fruit.reagents.total_volume)
+			var/amount = fruit.seed.potency / 2
+			if(fruit.distill_reagent)
+				reagents.add_reagent(fruit.distill_reagent, amount)
+				qdel(fruit)
+				return
+			else
+				var/data = list()
+				data["name"] = "[initial(fruit.name)] wine"
+				data["color"] = fruit.filling_color
+				data["boozepwr"] = fruit.wine_power
+				data["taste"] = fruit.tastes[1]
+				if(fruit.wine_flavor)
+					data["taste"] = fruit.wine_flavor
+				var/power_desc = "flavorful"
+				if(data["boozepwr"] >= 15)
+					power_desc = "mild"
+				if(data["boozepwr"] >= 30)
+					power_desc = "rich"
+				if(data["boozepwr"] >= 60)
+					power_desc = "strong"
+				if(data["boozepwr"] >= 80)
+					power_desc = "very strong"
+				if(data["boozepwr"] >= 120) //Should really only happen with strange plants...
```

Comment:
This whole if wall is wrong
If you checked the thing to be less powerful than 15 than you dont need to check it to be less powerful than 30 and so on

Reply:
It's being replaced anyway! Should have mentioned.